### PR TITLE
Binary Record Fixe

### PR DESCRIPTION
**Pull request by cs2be**
* Fixed issues with binary_record.h filename generator method.  std::hash is not consistent across OS.  It will generate different hash's when ran on different OS on the same string.  Change the filename generator to use a static counter.  It will be more ideal to use UUID generator available in boost 1.4.2, however, boost 1.4.2 has problems compiling on c++11 

* Fix issue with pytorch demo where it logs to the absolute path /workspace

* Fix performance issue on server when server fails to load image file.  Currently server retries an API call 3 times before it fails, at which point it'll sleep the thread for 2 secs.  This becomes problematic when there are alot of SDK errors.  I'll create seperate issue to refine the retry logic to not be so expensive on the server.

### CODE REVIEW

File: visualdl/storage/binary_record.h [hunk 1]
```diff
@@ -52,42 +51,59 @@ struct BinaryRecord {
     file.write(data_.data(), data_.size());
   }
 
-  const std::string& hash() { return hash_; }
+  static std::string generateFileName() {
```
Comment (Superjomn) [inline visualdl/storage/binary_record.h]: GenerateFileName

according to https://google.github.io/styleguide/cppguide.html#Function_Names

File: visualdl/storage/binary_record.h [hunk 2]
```diff
@@ -52,42 +51,59 @@ struct BinaryRecord {
...
+    // empty file. For now we don't throw an assert since it causes performance
+    // problems on the server.
+    // TODO(thuan): Update SDK.cc to not add image record if user stops training
+    // abruptly and no filename_ is set.
+    // TODO(thuan): Optimize visualDL server retry logic when a request fails.
+    // Currently if SDK call fails, server will issue multiple retries,
+    // which impacts performance.
+    if (!filename_.empty()) {
+      std::string path = dir_ + "/" + filename_;
+      std::cout << "READING FILE: " << path << std::endl;
```
Comment (Superjomn) [inline visualdl/storage/binary_record.h]: remove this line, in the training period, `cout` with an `endl` is low performance.

Comment (cs2be) [reply visualdl/storage/binary_record.h]: Ops sorry, forgot to remove this line, will remove it.

File: visualdl/storage/binary_record.h (hunk 1 above)
Comment (cs2be) [reply visualdl/storage/binary_record.h]: Will do

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**